tensorboard/data/server/reservoir.rs

```diff
@@ -413,13 +518,47 @@ mod tests {
     fn test_empty() {
         let mut rsv = StageReservoir::new(0);
         let mut head = Basin::new();
-        for i in 0..100 {
+        for mut i in 0..100 {
+            if i > 60 {
+                i -= 20; // "preemption", but not really, since no records
+            }
             rsv.offer(Step(i), ());
+            assert!(!rsv.staged_preemption());
             assert_eq!(rsv.staged_items(), &[]);
             if i % 5 == 0 {
                 rsv.commit(&mut head);
                 assert_eq!(head.as_slice(), &[]);
             }
         }
     }
+
+    #[test]
+    fn test_preempt_to_start() {
+        let mut rsv = StageReservoir::new(100);
+        let mut head = Basin::new();
+        for i in 0..100_000 {
```

Comment:
I gather that this loop over 100,000 random samples is fast enough to not really be noticeable when running tests?  (I assume the point of going all the way up to 100,000 is to minimize the chance of the test passing by accident if we hadn't reset `self.seen` all the way to 0.)